apps/docs/content/guides/functions.mdx

```diff
@@ -14,6 +14,40 @@ Edge Functions are server-side TypeScript functions, distributed globally at the
 - It is TypeScript first and supports WASM.
 - Edge Functions are globally distributed for low-latency.
 
+## How it works
+
+- **Request enters an edge gateway (relay)** — the gateway routes traffic, handles auth headers/JWT validation, and applies routing/traffic rules.
+- **Auth & policies are applied** — the gateway (or your function) can validate Supabase JWTs, apply rate-limits, and centralize security checks before executing code.
+- **Deno edge runtime executes your function** — the function runs on a regionally-distributed Deno node closest to the user for minimal latency.
+- **Integrations & data access** — functions commonly call Supabase APIs (Auth, Postgres, Storage) or third-party APIs. For Postgres, prefer connection strategies suited for edge/serverless environments (see the `connect-to-postgres` guide).
+- **Observability and logs** — invocations emit logs and metrics you can explore in the dashboard or downstream monitoring (Sentry, etc.).
+- **Response returns via the gateway** — the gateway forwards the response back to the client and records request metadata.
+
+## Quick technical notes
+
+- **Runtime:** Deno (TypeScript-first). Functions are simple `.ts` files that export a handler.
+- **Local dev parity:** `supabase functions serve` provides a local runtime similar to production for faster iteration.
+- **Global deployment:** `supabase functions deploy` publishes functions to the edge network.
+- **Cold starts & concurrency:** cold starts are possible — design for short-lived, idempotent operations. Heavy long-running jobs should be moved to background workers.
+- **Database connections:** treat Postgres like a remote, pooled service — use connection pools or serverless-friendly drivers. See `/guides/functions/connect-to-postgres`.
+- **Secrets:** store credentials in Supabase project secrets and access them via environment variables.
+
+## When to use Edge Functions
+
+- Authenticated or public HTTP endpoints that need low latency.
+- Webhook receivers (Stripe, GitHub, etc.).
+- On-demand image or Open Graph generation.
+- Small AI inference tasks or orchestrating calls to external ML APIs.
+- Transformations, proxying, and facade endpoints for third-party services.
+
+## Best practices
```

Comment:
Letr's drop the Best practices section. It's hard to understand for a new user and is not actionable. These Best practices are best taught with examples, which we have later in other guides